test/systemtest/conftest.py

```diff
@@ -27,6 +28,25 @@ def pytest_exception_interact(node, call, report):
 
     utils.dump_temp_files(node.funcargs['tmp_path_factory'].getbasetemp())
 
+
+@pytest.fixture(scope="session")
+def localconf(request):
+    """Host-local configuration."""
+    if os.getenv('OPENTITAN_TEST_LOCALCONF') and os.path.isfile(
+            os.environ['OPENTITAN_TEST_LOCALCONF']):
+        localconf_yaml_file = os.environ['OPENTITAN_TEST_LOCALCONF']
+    else:
+        XDG_CONFIG_HOME = os.getenv(
+            'XDG_CONFIG_HOME', os.path.join(os.environ['HOME'], '.config'))
+        localconf_yaml_file = os.path.join(XDG_CONFIG_HOME, 'opentitan',
+                                           'test-localconf.yaml')
+    logging.getLogger(__name__).info('Reading configuration from ' +
+                                     localconf_yaml_file)
+
+    with open(str(localconf_yaml_file), 'r') as fp:
+        yield yaml.load(fp, Loader=yaml.SafeLoader)
```

Comment:
Does this actually need to `yield` rather than `return`?

Reply:
I'm not sure that's right. Surely, `fp` would get closed as `return` unwound the lexical scope past the `with`. Using yield holds the file open longer, right?

Reply:
Oh, I got confused there and take that statement back. (I thought we return `fp`, but we actually something else.) So return is sufficient, and I changed to it.